core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -43,6 +43,17 @@ abstract class RawModule(implicit moduleCompileOptions: CompileOptions) extends
 
   val compileOptions = moduleCompileOptions
 
+  // This could be factored into a common utility
+  private def canBeNamed(id: HasId): Boolean = id match {
+    case d: Data =>
+      d.binding match {
+        case Some(_: ConstrainedBinding) => true
+        case _ => false
+      }
+    case b: BaseModule => true
+    case m: Mem[_]     => true
```

Comment:
can Instances be named?

Reply:
does this need a _ default case? What about Verification statements?

Reply:
Yes and yes, will fix and add tests